### PR TITLE
front: Add initial AuthenticationScreen

### PR DESCRIPTION
**Pull request by satanja**
The authentication screen.

### CODE REVIEW

File: front/src/AuthenticationScreen/index.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import PropTypes from "prop-types";
+
+/**
+ * An authentication screen. Upon a submit it will send the token in the inputfield
+ * along with a function to handle the response message. Authentication screen is
+ * responsible for its unrendering.
+ * Author: S.A. Tanja
+ *
+ * Props:
+ *  - handleSubmit: A functional component that will send the token to the SRRS
```
Comment (helaan) [inline front/src/AuthenticationScreen/index.js]: Why not call it onSubmit?

File: front/src/AuthenticationScreen/index.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ */
+class AuthenticationScreen extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: "",
+      handleSubmit: token => props.handleSubmit(token, this.handleResponse)
```
Comment (helaan) [inline front/src/AuthenticationScreen/index.js]: Ewww. Just use props directly

File: front/src/AuthenticationScreen/index.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleResponse = this.handleResponse.bind(this);
```
Comment (helaan) [inline front/src/AuthenticationScreen/index.js]: Why is this needed?

File: front/src/AuthenticationScreen/index.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  render() {
+    const token = this.state.value;
+    return (
+      <form onSubmit={() => this.state.handleSubmit(token)}>
+        <label>
+          Name:
+          <input
+            type="text"
+            value={this.state.value}
+            onChange={this.handleChange}
```
Comment (helaan) [inline front/src/AuthenticationScreen/index.js]: If you change this to () => this.handleChange(), you don't need to bind in the constructor.

File: front/src/AuthenticationScreen/index.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  handleChange(event) {
+    this.setState({ value: event.target.value });
+  }
+
+  /**
+   * In case of a incorrect token, update state to display error. In case of a correct
+   * token, update state to render null.
+   */
+  handleResponse(message) {
+    window.alert(message);
```
Comment (helaan) [inline front/src/AuthenticationScreen/index.js]: I thought you found an use for NotificationWindow, now you just use alert.

File: front/src/AuthenticationScreen/index.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+  render() {
+    const notification = {
+      message: "Incorrect token.",
+      notificationType: NotificationTypes.ERROR
+    };
+    return (
+      <div>
+        {this.state.loggedIn ? null : (
+          <form onSubmit={event => this.onSubmit(event)}>
```
Comment (helaan) [inline front/src/AuthenticationScreen/index.js]: Do you even need a form? When not just a field an a button, then you can leave out some form specific things

File: front/src/AuthenticationScreen/index.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+      this.onResponse(isCorrectToken)
+    );
+    // Suppress refresh of the application
+    event.preventDefault();
+  }
+
+  /**
+   * Update state to unrender the "Incorrect token" notification.
+   */
+  onDismiss() {
```
Comment (helaan) [inline front/src/AuthenticationScreen/index.js]: consider inlining small functions like this one that are only used once

File: front/src/AuthenticationScreen/index.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
+import React, { Component } from "react";
+import NotificationWindow from "../NotificationWindow/";
```
Comment (helaan) [inline front/src/AuthenticationScreen/index.js]: I don't think you need the trailing slash here

File: front/src/AuthenticationScreen/index.js [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+              type="text"
+              value={this.state.value}
+              onChange={event => this.onChange(event)}
+            />
+            <input type="submit" value="Submit" />
+          </form>
+        )}
+        {this.state.showNotification && (
+          <NotificationWindow
+            notification={notification}
```
Comment (helaan) [inline front/src/AuthenticationScreen/index.js]: consiider inlining the notification

File: front/src/AuthenticationScreen/index.js [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+    }
+  }
+
+  /**
+   * Gets the user inputed token, and calls the onSubmit function given from
+   * props, and gives a callback to update state.
+   */
+  onSubmit(event) {
+    const token = this.state.token;
+    this.props.onSubmit(token, isCorrectToken =>
```
Comment (helaan) [inline front/src/AuthenticationScreen/index.js]: Should we use callbacks here or promises?

File: front/src/AuthenticationScreen/index.js [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+    super(props);
+    this.state = {
+      token: "",
+      showNotification: false,
+      loggedIn: false
+    };
+  }
+
+  /**
+   * Callback function. In case of a incorrect token, update state to display
```
Comment (gwvdh) [inline front/src/AuthenticationScreen/index.js]: *an

File: front/src/AuthenticationScreen/index.js [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+   * error. In case of a correct token, update state to render null.
+   * @param isCorrectToken whether the token sent is the correct token to log in
+   */
+  onResponse(isCorrectToken) {
+    if (!isCorrectToken) {
+      this.setState({ showNotification: true });
+    } else {
+      this.setState({ showNotification: false, loggedIn: true });
```
Comment (gwvdh) [inline front/src/AuthenticationScreen/index.js]: Why not 'if(isCorrectToken)' and swap what happens

File: front/src/AuthenticationScreen/index.js [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+    event.preventDefault();
+  }
+
+  render() {
+    return (
+      <div>
+        {this.state.loggedIn ? null : (
+          <form onSubmit={event => this.onSubmit(event)}>
+            <label>TRC Token:</label>
+            <input
```
Comment (gwvdh) [inline front/src/AuthenticationScreen/index.js]: No styling?

File: front/src/App.js [hunk 14]
```diff
@@ -135,47 +138,75 @@ class App extends Component {
     });
   }
 
+  /**
+   * The function AuthenticationScreen will call when a token has been
+   * submitted.
+   * @param token The token gotten from AuthenticationScreen.
```
Comment (helaan) [inline front/src/App.js]: gotten -> received?

File: front/src/App.js [hunk 15]
```diff
@@ -135,47 +138,75 @@ class App extends Component {
...
+              {activePage === "refbox" && (
+                <div>
+                  <RefboxField isPenalty={false} />
+                  <RefboxSettings />
+                </div>
+              )}
+              {activePage === "settings" && (
+                <Fragment>
+                  <TurtleEnableBar
+                    turtles={turtles.map(turtle => {
```
Comment (helaan) [inline front/src/App.js]: This line was changed, please update

File: front/src/App.js [hunk 16]
```diff
@@ -135,47 +138,75 @@ class App extends Component {
...
+                        enabled: turtle.enabled
+                      };
+                    })}
+                    onTurtleEnableChange={position =>
+                      this.onTurtleEnableChange(position)
+                    }
+                  />
+                  <ScrollableContent>
+                    <Settings
+                      turtles={turtles.filter(turtle => turtle.enabled)}
```
Comment (helaan) [inline front/src/App.js]: This line was changed, please update

File: front/src/App.js [hunk 17]
```diff
@@ -135,47 +138,75 @@ class App extends Component {
...
+              {activePage === "settings" && (
+                <Fragment>
+                  <TurtleEnableBar
+                    turtles={turtles.map(turtle => {
+                      return {
+                        id: turtle.id,
+                        enabled: turtle.enabled
+                      };
+                    })}
+                    onTurtleEnableChange={position =>
```
Comment (helaan) [inline front/src/App.js]: position -> id, please change

File: front/src/App.js [hunk 18]
```diff
@@ -138,49 +140,71 @@ class App extends Component {
...
 
+  /**
+   * The function AuthenticationScreen will call when a token has been
+   * submitted.
+   * @param token The token received from AuthenticationScreen.
+   * @param onIncorrectToken Callback from AuthenticationScreen to update the
+   * AuthenticationScreen in case the token was incorrect.
+   */
+  onSubmit(token, onIncorrectToken) {
+    if (token === "techunited") {
```
Comment (gwvdh) [inline front/src/App.js]: Add a TODO

File: front/src/AuthenticationScreen/index.js [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+          </LoginButton>
+        </Window>
+      </Container>
+    );
+  }
+}
+
+//TODO: Apply theme.js for all styled components (if applicable)
+const Container = styled.div`
+  background: silver;
```
Comment (gwvdh) [inline front/src/AuthenticationScreen/index.js]: theme.js?
